src/client/runner/automation/playback/click/index.js

```diff
@@ -0,0 +1,274 @@
+                var activeElement         = domUtils.getActiveElement();
+
+                // NOTE: in IE focus is not raised if element was focused
+                // before click, even if focus is lost during mousedown
+                this.eventState.simulateFocus = !browserUtils.isIE || activeElement !== this.eventArgs.element ||
+                                                isContentEditableBody;
+
+                // NOTE: In WebKit and IE, the mousedown event opens the select element's dropdown;
+                // therefore, we should prevent mousedown and hide the dropdown (B236416).
+                var needCloseSelectDropDown = (browserUtils.isWebKit || browserUtils.isIE) &&
+                                              domUtils.isSelectElement(this.element);
+
+                if (needCloseSelectDropDown)
+                    this._bindMousedownHandler();
+
+                this._bindBlurHandler(activeElement);
+
+                this.eventState.simulateDefaultBehavior =
+                    eventSimulator.mousedown(this.eventArgs.element, this.eventArgs.options);
+
+                if (this.eventState.simulateDefaultBehavior === false) {
+                    this.eventState.simulateDefaultBehavior = needCloseSelectDropDown &&
+                                                              !this.eventState.mousedownPrevented;
+                }
+
+                return this._ensureActiveElementBlur(activeElement);
+            })
+            .then(() => this._focus())
+            .then(() => delay(automationSettings.ACTION_STEP_DELAY));
+    }
+
+    _ensureActiveElementBlur (element) {
+        // NOTE: In some cases, mousedown may lead to active element change (browsers raise blur).
+        // We simulate the blur event if the active element was changed after the mousedown, and
+        // the blur event does not get raised automatically (B239273, B253520)
+        return new Promise(resolve => {
+            var simulateBlur = domUtils.getActiveElement() !== element && !this.eventState.blurRaised;
+
+            if (!simulateBlur) {
+                resolve();
+                return;
+            }
+
+            if (browserUtils.isIE && browserUtils.version < 12) {
+                // NOTE: In whatever way an element is blurred from the client script, the
+                // blur event is raised asynchronously in IE (in MSEdge focus/blur is sync)
+                nextTick()
+                    .then(() => {
+                        if (!this.eventState.blurRaised)
+                            eventSimulator.blur(element);
+
+                        resolve();
+                    });
+            }
+            else {
+                eventSimulator.blur(element);
+                resolve();
+            }
+        });
+    }
+
+    _focus () {
+        if (this.eventState.simulateDefaultBehavior === false)
+            return delay(automationSettings.ACTION_STEP_DELAY);
```

Comment:
as I see we have doubled waiting in this case, here and in [`_mousedown`](https://github.com/DevExpress/testcafe/pull/387/files#diff-a64ce8c2274ce9c6c74023630e34c4deR173)